### ISSUE TITLE
ZTEST Framework fails when ztest_run_all is called multiple times

### ISSUE THREAD

**Issue by asemjonovs** [zephyrproject-rtos/zephyr#48424]
**Describe the bug**
When overriding test_main to run ZTEST framework multiple times a failure status may be returned as the fail flag will be set early on
 in `ztest_verify_all_test_suites_ran`

```
      /* Only suites that use a predicate checking for phase == PWR_PHASE_0 will run. */
      state.phase = PWR_PHASE_0;
      ztest_run_all(&state); // test_status set to failure here

      /* Only suites that use a predicate checking for phase == PWR_PHASE_1 will run. */
      state.phase = PWR_PHASE_1;
      ztest_run_all(&state); // all tests are confirmed to have ran but test_status failure won't be cleared
```

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using? native_posix
 - What have you tried to diagnose or workaround this issue?
 - ...

**To Reproduce**
1. See above bug description

**Expected behavior**
Expect a passing status returned when no tests fail and all tests have been executed.

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)
annoyance

**Logs and console output**
If applicable, add console logs or other types of debug information
e.g Wireshark capture or Logic analyzer capture (upload in zip archive).
copy-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue. (if unable to obtain text log, add a screenshot)

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows): Linux
 - Toolchain (e.g Zephyr SDK, ...): Zephyr
 - Commit SHA or Version used: latest

**Additional context**
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...

**Comment by henrikbrixandersen** [comment]
@asemjonovs Was this resolved in https://github.com/zephyrproject-rtos/zephyr/pull/48430?

**Comment by asemjonovs** [comment]
> @asemjonovs Was this resolved in #48430?

Yes, this is resolved with above.
